src/containerDeployment/dockerUtils.ts

```diff
@@ -0,0 +1,688 @@
+            bodyText: ContainerDeployment.settingUpContainerBody,
+            stepAction: checkIfContainerIsReadyForConnections,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: [],
+            headerText: ContainerDeployment.connectingToContainerHeader,
+            bodyText: ContainerDeployment.connectingToContainerBody,
+            stepAction: undefined,
+        },
+    ];
+}
+
+/**
+ * Handles the result of a Docker command and updates the corresponding step statuses accordingly.
+ */
+export function setStepStatusesFromResult(
+    result: DockerCommandParams,
+    currentStep: DockerStepOrder,
+    steps: DockerStep[],
+): DockerStep[] {
+    if (result.success) {
+        steps[currentStep].loadState = ApiStatus.Loaded;
+    } else {
+        steps[currentStep].loadState = ApiStatus.Error;
+        steps[currentStep].errorMessage = result.error;
+        steps[currentStep].fullErrorText = truncateAndSanitizeErrorTextIfNeeded(
+            result.fullErrorText,
+        );
+        for (let i = currentStep + 1; i < steps.length; i++) {
+            steps[i].loadState = ApiStatus.Error;
+            steps[i].errorMessage = ContainerDeployment.previousStepFailed;
+        }
+    }
+    return steps;
+}
+
+/**
+ * Truncates the error text if it exceeds the maximum length and sanitizes sensitive info.
+ */
+export function truncateAndSanitizeErrorTextIfNeeded(errorText: string): string {
+    // Sanitize
+    const sanitized = errorText.replace(/(SA_PASSWORD=)([^ \n]+)/gi, '$1******"');
+
+    // Truncate if necessary
+    if (sanitized.length > MAX_ERROR_TEXT_LENGTH) {
+        return `${sanitized.substring(0, MAX_ERROR_TEXT_LENGTH)}...`;
+    }
+
+    return sanitized;
+}
+
+/**
+ * Checks if the SQL Server password meets the complexity requirements.
+ * If the password is valid, it returns the validation message, which is an empty string.
+ * If the password is invalid, it returns an error message.
+ */
+export function validateSqlServerPassword(password: string): string {
+    if (password.length < 8) {
+        return ContainerDeployment.passwordLengthError;
+    }
+
+    const hasUpperCase = /[A-Z]/.test(password);
+    const hasLowerCase = /[a-z]/.test(password);
+    const hasDigit = /\d/.test(password);
+    const hasSpecialChar = /[!@#$%^&*]/.test(password);
+
+    // Count the number of required character categories met
+    const categoryCount = [hasUpperCase, hasLowerCase, hasDigit, hasSpecialChar].filter(
+        Boolean,
+    ).length;
+
+    if (categoryCount < 3) {
+        return ContainerDeployment.passwordComplexityError;
+    }
+
+    return "";
+}
+
+/**
+ * Checks if the provided connection name is valid and not a duplicate connection name within
+ * mssql.connections
+ */
+export function validateConnectionName(connectionName: string): boolean {
+    const connections = vscode.workspace
+        .getConfiguration(extensionName)
+        .get(connectionsArrayName, []);

```

Comment:
NIT: Use connection manager to read connection. This is a little prone to breaking. It also doesn't get global connections. So if someone has connections in the workspace you don't read global values.